### PR TITLE
em100: init at 0-unstable-2024-11-14

### PR DESCRIPTION
**Pull request by felixsinger**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by funkeleinhorn — APPROVED** [review]
LGTM. Tested build on x86 Linux and checked that the binary looks alright. Testing the functionality is not possible for me as of missing hardware.

File: pkgs/by-name/em/em100/package.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [
+    curl
+    libusb1
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    make em100
```
Comment (Atemu) [inline pkgs/by-name/em/em100/package.nix:29]: This breaks e.g. parallel building (which you should probably also enable) and other tweaks we have in the regular buildPhase.

Please use `buildFlags` instead.

File: pkgs/by-name/em/em100/package.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  stdenv,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "em100";
+  version = "unstable-2024-11-14";
+
+  src = fetchgit {
+    url = "https://review.coreboot.org/em100";
+    rev = "a78b4ba4774f05ecd7af495604b437113596d70e";
```
Comment (Atemu) [inline pkgs/by-name/em/em100/package.nix:16]: Where does this rev come from? Please document e.g. the branch this is taken from.

**Review by Atemu — APPROVED** [review]
LGTM apart from these smaller things

**Review by Atemu — APPROVED** [review]
Diff LGTM does everything still work as expected?

**Comment by felixsinger** [comment]
> Diff LGTM does everything still work as expected?

Yeah, the build flag works as intended and the binary works too. Thanks :)

**Pull request merged**